### ISSUE TITLE
lifecycle_rule on aws_s3_bucket causes terraform to think output changes

### ISSUE THREAD

**Issue by archoversight** [hashicorp/terraform-provider-aws#16615]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.12.0
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/hashicorp/time v0.6.0
```

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_s3_bucket" "special_keys" {
  bucket = "my-special-bucket"
  acl    = "private"
  tags   = {}

  versioning {
    enabled = true
  }

  lifecycle_rule {
    id      = "cleanup_versions"
    enabled = true

    noncurrent_version_transition {
      days          = 30
      storage_class = "STANDARD_IA"
    }

    noncurrent_version_transition {
      days          = 60
      storage_class = "GLACIER"
    }

    noncurrent_version_expiration {
      days = 90
    }
  }
}

output "s3_special_keys" { value = aws_s3_bucket. special_keys }

# Workaround
output "s3_special_keys_work_around" { value = { for k, v in aws_s3_bucket.special_keys : k => v if k != "lifecycle_rule" } }
```

### Debug Output

```
s3_special_keys = {
        acceleration_status                  = ""
        acl                                  = "private"
        arn                                  = ""
        bucket                               = "my-special-bucket"
        bucket_domain_name                   = "my-special-bucket.s3.amazonaws.com"
        bucket_prefix                        = null
        bucket_regional_domain_name          = ""
        cors_rule                            = []
        force_destroy                        = false
        grant                                = []
        hosted_zone_id                       = ""
        id                                   = "my-special-bucket-id"
        lifecycle_rule                       = [
            {
                abort_incomplete_multipart_upload_days = 0
                enabled                                = true
                expiration                             = []
                id                                     = "versioned_backup"
                noncurrent_version_expiration          = [
                    {
                        days = 90
                    },
                ]
                noncurrent_version_transition          = [
                    {
                        days          = 30
                        storage_class = "STANDARD_IA"
                    },
                    {
                        days          = 60
                        storage_class = "GLACIER"
                    },
                ]
                prefix                                 = ""
                tags                                   = {}
                transition                             = []
            },
        ]
        logging                              = []
        object_lock_configuration            = []
        policy                               = null
        region                               = "us-gov-special"
        replication_configuration            = []
        request_payer                        = "BucketOwner"
        server_side_encryption_configuration = []
        tags                                 = {}
        versioning                           = [
            {
                enabled    = true
                mfa_delete = false
            },
        ]
        website                              = []
        website_domain                       = null
        website_endpoint                     = null
    }
```

### Expected Behavior

Terraform shouldn't tell me the output has changed, because it hasn't.

### Actual Behavior

Terraform says there are changes to the output

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

### Important Factoids

This was in govCloud, haven't tested against AWS commercial.

### References

* #0000

**Comment by bill-rich** [comment]
Hi @archoversight! Please upgrade to v0.14.2 and give this another try. There were some changes with the the recent releases to address similar issues https://github.com/hashicorp/terraform/releases/tag/v0.14.1.

**Comment by archoversight** [comment]
@bill-rich sadly due to other dependencies I am currently unable to test if this is fixed in 0.14.2 as I can't upgrade from the 0.13.x branch yet.
